app/models/location.rb

```diff
@@ -56,4 +63,27 @@ def self.find_or_create_by_api_ids(locationiq_id, osm_id, osm_type)
       create_from_params(resp)
     end
   end
+
+  # Restrict Human location specificity to State, Country. Return new Location if restriction added.
+  def self.check_and_restrict_specificity(location, host_genome_name)
+    # We don't want Human locations with subdivision or city
+    if host_genome_name == "Human" && (location.subdivision_name.present? || location.city_name.present?)
+      # Redo the search for just the country/state
+      success, resp = geosearch_by_country_and_state(location.country_name, location.state_name)
+      unless success
+        raise "Couldn't find #{location.state_name}, #{location.country_name} (state, country)"
+      end
+
+      result = LocationHelper.adapt_location_iq_response(resp[0])
+      existing = Location.find_by(locationiq_id: result[:locationiq_id])
+      new_location = existing ? existing : create_from_params(result)
+
+      # Delete the overly specific location if not used anymore
+      location.delete if Metadatum.where(location: location).empty?
```

Comment:
Why is this the case? We are temporarily saving it before?

Reply:
I agree it would be better to never save it to the database in the first place. (BTW, deleted rows are still retained in fivetran)

Reply:
@cdebourcy how does that work? Only if they deleted after the fivetran sync, right?

Reply:
Yes -- the point is, it's possible. As you allude to, it also just seems bad form to create a record only to delete it later (there is actually another metadata endpoint that does something similar, which really bugs me as well...)

Reply:
Thanks, just wanted to understand we are talking about the same case.
Which endpoint is that?

Reply:
refactored 'create_from_params' to just 'new_from_params' to avoid this 👍